src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,503 @@
+// mailbox_channel_switch = 0
+// xclbin_change = 1
+// cahce_xclbin = 0
+// we may support in the future, like,
+// [Daemon]
+// host_ip = x.x.x.x
+static void load_config(const std::shared_ptr<xrt_core::device>& _dev, const std::string path)
+{
+  boost::property_tree::ptree pt_root;
+  boost::property_tree::ini_parser::read_ini(path, pt_root);
+  static boost::property_tree::ptree empty_tree;
+
+  const boost::property_tree::ptree pt_device = pt_root.get_child("Device", empty_tree);
+
+  if (pt_device.empty())
+    throw std::runtime_error(boost::str(boost::format("No [Device] section in the config file. Config File: %s") % path));
+
+  for (auto& key : pt_device) {
+    if (!key.first.compare("mailbox_channel_disable")) {
+      xrt_core::device_update<xrt_core::query::config_mailbox_channel_disable>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("mailbox_channel_switch")) {
+      xrt_core::device_update<xrt_core::query::config_mailbox_channel_switch>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("xclbin_change")) {
+      xrt_core::device_update<xrt_core::query::config_xclbin_change>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("cache_xclbin")) {
+      xrt_core::device_update<xrt_core::query::cache_xclbin>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("scaling_enabled")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_enabled>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("scaling_power_override")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_power_override>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("scaling_temp_override")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_temp_override>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    throw std::runtime_error(boost::str(boost::format("'%s' is not a supported config entry") % key.first));
+  }
+}
+
+static config
+get_daemon_conf()
+{
+  config cfg;
+  cfg.host = xrt_core::get_hostname();
+
+  std::ifstream istr(config_file);
+  if (!istr)
+    return cfg;
+
+  // Load persistent value, may overwrite default one
+  for (std::string line; std::getline(istr, line);) {
+    auto pos = line.find('=', 0);
+    if (pos == std::string::npos)
+      throw xrt_core::system_error(EIO, "Bad daemon config file line '" + line + "'");
+    auto key = line.substr(0, pos++);
+    auto value = line.substr(pos);
+
+    if (key == "host") 
+      cfg.host = value;
+  }
+
+  return cfg;
+}
+
+/*
+ * helper function for option: showx
+ * shows daemon config
+ */
+static void 
+show_daemon_conf()
+{
+  auto cfg = get_daemon_conf();
+  std::cout << "Daemon:" << std::endl;
+  std::cout << "\t";
```

Comment:
This output should be using boost::format and we should especially not be using tabs.

Reply:
Removed the write function and replaced it with an operator<< function to allow usage with boost::format